packages/blade/src/components/Alert/Alert.tsx

```diff
@@ -139,14 +141,37 @@ const Alert = ({
       '[Blade: Alert]: SecondaryAction is allowed only when PrimaryAction is defined.',
     );
   }
+  const { theme } = useTheme();
+  const { matchedDeviceType } = useBreakpoint({ breakpoints: theme.breakpoints });
+  const isMobile = matchedDeviceType === 'mobile';
+
   const [isVisible, setIsVisible] = useState(true);
   const contrastType = `${contrast}Contrast` as const;
   const iconSize = isBorderless ? 'large' : 'medium';
   const textSize = isBorderless ? 'medium' : 'small';
 
   const Icon = intentIconMap[intent];
+  let iconOffset: DotNotationSpacingStringToken = 'spacing.1';
+
+  // certain special cases below needs special care for near perfect alignment
+  if (isReactNative) {
+    if (isBorderless && !title) {
+      iconOffset = 'spacing.1';
+    } else if (!isBorderless && !title) {
+      iconOffset = 'spacing.0';
+    } else if (!isBorderless && title) {
+      iconOffset = 'spacing.2';
+    }
+  } else if (isMobile) {
+    if (!isBorderless && title) {
+      iconOffset = 'spacing.2';
+    } else if (isBorderless && !title) {
+      iconOffset = 'spacing.2';
+    }
+  }
+
   const icon = (
-    <Box marginTop="spacing.1" display="flex">
+    <Box marginTop={iconOffset} display="flex">
```

Comment:
why is `spacing.1` behaving weirdly in diff scenarios? will it be solved if we use `alingItems:center` or something?

Reply:
Various cases:
- the bordered version uses a text as title whereas the borderless one uses heading
- the font size increases slightly for mobile
- we also need to handle description only alerts which uses a different sized text (plus web and mobile layouts)
- react native isn't mapping 1:1 with web (I think platform differences)

Regarding center `alignItems` that would make the icon off center because we need to align it to top

Reply:
Damn. The logic is quite specific 😢 

Have you tried running on both Android & iOS to make sure the logic stands true?

Reply:
Yes, please check the screenshots in PR description, took quite a while 😅